### ISSUE TITLE
document data format expected by NetworkFrame

### ISSUE THREAD

**Issue by micahstubbs** [emeeks/semiotic#218]
it would be nice to document the data format expected by the NetworkFrame component

https://emeeks.github.io/semiotic/#/semiotic/networkframe

**Comment by micahstubbs** [comment]
in the examples, this format seems to be used for edges

```
export const edgeData = [
  { source: "Jim", target: "Irene", weight: 5 },
  { source: "Susie", target: "Irene", weight: 5 },
  { source: "Jim", target: "Susie", weight: 5 },
  { source: "Susie", target: "Kai", weight: 5 },
  { source: "Shirley", target: "Kai", weight: 5 },
  { source: "Shelby", target: "Kai", weight: 5 },
  { source: "Kai", target: "Susie", weight: 5 },
  { source: "Kai", target: "Shirley", weight: 5 },
  { source: "Kai", target: "Shelby", weight: 5 },
  { source: "Erik", target: "Zan", weight: 5 },
  { source: "Tony", target: "Zan", weight: 5 },
  { source: "Tony", target: "Fil", weight: 5 },
  { source: "Tony", target: "Ian", weight: 5 },
  { source: "Tony", target: "Adam", weight: 5 },
  { source: "Fil", target: "Tony", weight: 4 },
  { source: "Ian", target: "Miles", weight: 1 },
  { source: "Adam", target: "Tony", weight: 3 },
  { source: "Miles", target: "Ian", weight: 3 },
  { source: "Erik", target: "Kai", weight: 2 },
  { source: "Erik", target: "Nadieh", weight: 2 },
  { source: "Jim", target: "Nadieh", weight: 2 },
  { source: "Elijah", target: "Enrico", weight: 2 },
  { source: "Enrico", target: "Moritz", weight: 2 },
  { source: "Moritz", target: "Enrico", weight: 2 },
  { source: "Moritz", target: "ET", weight: 2 },
  { source: "a", target: "b", weight: 2 },
  { source: "b", target: "c", weight: 2 },
  { source: "d", target: "e", weight: 2 },
  { source: "f", target: "g", weight: 2 },
  { source: "h", target: "i", weight: 2 }
]
```

source https://github.com/emeeks/semiotic/blob/master/src/docs/example_settings/networkframe.js

**Comment by micahstubbs** [comment]
![screen shot 2018-05-04 at 2 43 25 pm](https://user-images.githubusercontent.com/2119400/39654408-865d06d8-4fa9-11e8-9e7f-1abc42085028.png)

**Comment by micahstubbs** [comment]
as one graph object, the input data could look like this:

```json
{
  "nodes": [
    {
      "id": "Susie"
    },
    {
      "id": "Kai"
    },
    {
      "id": "Elijah"
    },
    {
      "id": "Enrico"
    },
    {
      "id": "j"
    },
    {
      "id": "k"
    },
    {
      "id": "l"
    }
  ],
  "edges": [
    {
      "source": "Jim",
      "target": "Irene",
      "weight": 5
    },
    {
      "source": "Susie",
      "target": "Irene",
      "weight": 5
    },
    {
      "source": "Jim",
      "target": "Susie",
      "weight": 5
    },
    {
      "source": "Susie",
      "target": "Kai",
      "weight": 5
    },
    {
      "source": "Shirley",
      "target": "Kai",
      "weight": 5
    },
    {
      "source": "Shelby",
      "target": "Kai",
      "weight": 5
    },
    {
      "source": "Kai",
      "target": "Susie",
      "weight": 5
    },
    {
      "source": "Kai",
      "target": "Shirley",
      "weight": 5
    },
    {
      "source": "Kai",
      "target": "Shelby",
      "weight": 5
    },
    {
      "source": "Erik",
      "target": "Zan",
      "weight": 5
    },
    {
      "source": "Tony",
      "target": "Zan",
      "weight": 5
    },
    {
      "source": "Tony",
      "target": "Fil",
      "weight": 5
    },
    {
      "source": "Tony",
      "target": "Ian",
      "weight": 5
    },
    {
      "source": "Tony",
      "target": "Adam",
      "weight": 5
    },
    {
      "source": "Fil",
      "target": "Tony",
      "weight": 4
    },
    {
      "source": "Ian",
      "target": "Miles",
      "weight": 1
    },
    {
      "source": "Adam",
      "target": "Tony",
      "weight": 3
    },
    {
      "source": "Miles",
      "target": "Ian",
      "weight": 3
    },
    {
      "source": "Erik",
      "target": "Kai",
      "weight": 2
    },
    {
      "source": "Erik",
      "target": "Nadieh",
      "weight": 2
    },
    {
      "source": "Jim",
      "target": "Nadieh",
      "weight": 2
    },
    {
      "source": "Elijah",
      "target": "Enrico",
      "weight": 2
    },
    {
      "source": "Enrico",
      "target": "Moritz",
      "weight": 2
    },
    {
      "source": "Moritz",
      "target": "Enrico",
      "weight": 2
    },
    {
      "source": "Moritz",
      "target": "ET",
      "weight": 2
    },
    {
      "source": "a",
      "target": "b",
      "weight": 2
    },
    {
      "source": "b",
      "target": "c",
      "weight": 2
    },
    {
      "source": "d",
      "target": "e",
      "weight": 2
    },
    {
      "source": "f",
      "target": "g",
      "weight": 2
    },
    {
      "source": "h",
      "target": "i",
      "weight": 2
    }
  ]
}
```

**Comment by emeeks** [comment]
Yeah, I need more documentation. `NetworkFrame` has a `sourceAccessor` and `targetAccessor` that default, respectively, to `source` and `target` attributes and which correspond either to a string value that exists (or if not a node is generated) in the `nodes` data or to an actual object ref. You can adjust the node ID attribute via `nodeIDAccessor` which defaults to `id`.

I could see `NetworkFrame` accepting a `graph` property, that's pretty standard in the network viz world and would let somebody have a convenient way of sending data.

Note that `nodes` is always optional and `edges` is optional (for bubble packs) and that edges also accepts hierarchical JSON.

**Comment by emeeks** [comment]
The note in the hover on edge data is because there's no good way to parse the circular structure of the data so my parsing function just gives up.

**Comment by emeeks** [comment]
Resolved in 1.11.1
